qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -146,6 +145,15 @@ public void test041AmazonCaCertsAreInTheKeystore() {
         assertTrue("Expected Amazon trusted cert in cacerts", matches);
     }
 
+    /**
+     * Check that when the keystore is created on startup, it is created with the correct permissions.
+     */
+    public void test042KeystorePermissionsAreCorrect() throws Exception {
+        waitForElasticsearch(installation);
```

Comment:
The failures seem to indicate that `waitForElasticseach()` doesn't seem to be robust enough. Do we have theories on why that method would return, but the keystore file would _not_ be there? Wouldn't we expect this test to occasionally fail still?

Reply:
I'm not sure that `waitForElasticseach()` is the problem. The original problem in #66656 was due to to the test running a CLI tool without waiting for ES to start up and create the keystore, which in this case the tool needed.

I removed the keystore from the `verifyOssInstallation()` because most tests don't care about it, and have a test fail because it didn't wait for a file to be created that the test doesn't care about is just...silly.

Reply:
Sure, I guess my point being that the new test you added is still susceptible to the same underlying cause of those other failures, unless I'm missing something. Indeed I agree we don't need superfluous assertions in those other tests, so if nothing else this should fail _less_ often. We'll keep an eye on it.

Reply:
Basically this - if the new test is failing often, at least I have somewhere more specific to start investigating.